datafusion/core/src/datasource/file_format/csv.rs

```diff
@@ -494,56 +490,6 @@ impl CsvSink {
             file_compression_type,
         }
     }
-
-    // Create a write for Csv files
-    async fn create_writer(
```

Comment:
the whole point of this PR is to move this into a free function that is shared rather than have two copies

Reply:
Nice catch!